src/naive.py

```diff
@@ -1,16 +1,46 @@
 """Implementation of the naive exact matching algorithm."""
 
 import argparse
+import fasta
+import fastq
 
+def naive(read, genome):
+    length = len(read[1])
+    out = []
+    for i in range(len(genome[1]) - len(read[1]) + 1):
+        current_match_length = 0
```

Comment:
Except where you resent `current_match_length` to zero (which isn't necessary), this variable is identical to `j`. You might as well just use `j`.

You can also use a nifty feature in Python: `else`-blocks for loops. If you get through a loop without `break`'ing, the `else` part is run, but otherwise it isn't. So, you can use a `for` loop to run through the patter and break on a mismatch. If you have a mismatch, the `else` part isn't run; if you don't, it is.

```python
for j in range(len(read[1])):
    if genome[1][i + j] != read[1][j]:
        break
else:
    # You have a match at index i
```